lib/services/version-service.ts

```diff
@@ -2,41 +2,42 @@ import { join } from "path";
 import * as semver from "semver";
 
 export class VersionService implements IVersionService {
-	private static DEFAULT_CLI_VERSION = "3.0.0";
-
 	constructor(private $fs: IFileSystem,
 		private $httpClient: Server.IHttpClient,
-		private $logger: ILogger) { }
+		private $logger: ILogger,
+		private $projectDataService: IProjectDataService) { }
 
 	public async getCliVersion(runtimeVersion: string): Promise<string> {
 		try {
 			const latestMatchingVersion = process.env.TNS_CLI_CLOUD_VERSION || await this.getLatestMatchingVersion("nativescript", this.getVersionRangeWithTilde(runtimeVersion));
-			return latestMatchingVersion || VersionService.DEFAULT_CLI_VERSION;
+			if (!latestMatchingVersion) {
+				throw new Error(`Cannot find matching CLI version based on runtime version ${runtimeVersion}`);
+			}
+
+			return latestMatchingVersion;
 		} catch (err) {
 			this.$logger.trace(`Unable to get information about CLI versions. Error is: ${err.message}`);
+			if (!semver.valid(runtimeVersion)) {
+				// In case the runtime version is not a valid semver version, we cannot determine CLI version.
+				throw new Error("Unable to determine CLI version for cloud build.");
+			}
+
 			return `${semver.major(runtimeVersion)}.${semver.minor(runtimeVersion)}.0`;
```

Comment:
maybe we should remove this as well

Reply:
along with the whole `try ... catch` block. 
Let's just `throw` an error whenever there isn't a matching version and that's it